lib/node_modules/@stdlib/math/base/special/lnf/lib/main.js

```diff
@@ -0,0 +1,181 @@
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*
+* ## Notice
+*
+* The following copyright and license were part of the original implementation available as part of [FreeBSD]{@link https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/e_logf.c}. The implementation follows the original, but has been modified for JavaScript.
+*
+* ```text
+* Copyright (C) 1993 by Sun Microsystems, Inc. All rights reserved.
+*
+* Developed at SunPro, a Sun Microsystems, Inc. business.
+* Permission to use, copy, modify, and distribute this
+* software is freely granted, provided that this notice
+* is preserved.
+* ```
+*/
+
+'use strict';
+
+// MODULES //
+
+var toWordf = require( '@stdlib/number/float32/base/to-word' );
+var fromWordf = require( '@stdlib/number/float32/base/from-word' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var NINF = require( '@stdlib/constants/float32/ninf' );
+var FLOAT32_EXPONENT_MASK = require( '@stdlib/constants/float32/exponent-mask' );
+var FLOAT32_EXPONENT_BIAS = require( '@stdlib/constants/float32/exponent-bias' );
+var FLOAT32_ABS_MASK = require( '@stdlib/constants/float32/abs-mask' );
+var FLOAT32_SIGNIFICAND_MASK = require( '@stdlib/constants/float32/significand-mask' );
+var float64ToFloat32 = require( '@stdlib/number/float64/base/to-float32' );
+var polyvalP = require( './polyval_p.js' );
+var polyvalQ = require( './polyval_q.js' );
+
+
+// VARIABLES //
+
+var LN2_HI = 6.9313812256e-01; // 0x3f317180
+var LN2_LO = 9.0580006145e-06; // 0x3717f7d1
+var TWO25 = 3.355443200e+07;   // 0x4c000000
+var ONE_THIRD = 0.33333333333333333;
+
+
+// MAIN //
+
+/**
+* Evaluates the natural logarithm of a single-precision floating-point number.
+*
+* @param {NonNegativeNumber} x - input value
+* @returns {number} function value
+*
+* @example
+* var v = lnf( 4.0 );
+* // returns ~1.386
+*
+* @example
+* var v = lnf( 0.0 );
+* // returns -Infinity
+*
+* @example
+* var v = lnf( Infinity );
+* // returns Infinity
+*
+* @example
+* var v = lnf( NaN );
+* // returns NaN
+*
+* @example
+* var v = lnf( -4.0 );
+* // returns NaN
+*/
+function lnf( x ) {
+	var hfsq;
+	var ix;
+	var t2;
+	var t1;
+	var dk;
+	var k;
+	var R;
+	var f;
+	var i;
+	var j;
+	var s;
+	var w;
+	var z;
+
+	if ( x === 0.0 ) {
+		return NINF;
+	}
+	if ( isnan( x ) || x < 0.0 ) {
+		return NaN;
+	}
+	x = float64ToFloat32( x );
+	ix = toWordf( x );
+	k = 0;
+
+	// x < 2**-126
+	if ( ix < 0x00800000 ) {
+		if ( ( ix & FLOAT32_ABS_MASK ) === 0 ) {
+			// log(+-0)=-inf
+			return float64ToFloat32( -TWO25 / 0.0 );
+		}
```

Comment:
L119-121 already have a check for an input value of zero, so I don't think this is needed.